common4j/src/main/com/microsoft/identity/common/java/platform/AbstractDevicePopManager.java

```diff
@@ -100,48 +99,20 @@
-import static com.microsoft.identity.common.adal.internal.cache.StorageHelper.applyKeyStoreLocaleWorkarounds;
-import static com.microsoft.identity.common.java.util.ported.DateUtilities.LOCALE_CHANGE_LOCK;
-import static com.microsoft.identity.common.java.util.ported.DateUtilities.isLocaleCalendarNonGregorian;
-import static com.microsoft.identity.common.java.exception.ClientException.ANDROID_KEYSTORE_UNAVAILABLE;
-import static com.microsoft.identity.common.java.exception.ClientException.BAD_KEY_SIZE;
-import static com.microsoft.identity.common.java.exception.ClientException.BAD_PADDING;
-import static com.microsoft.identity.common.java.exception.ClientException.INTERRUPTED_OPERATION;
-import static com.microsoft.identity.common.java.exception.ClientException.INVALID_ALG;
-import static com.microsoft.identity.common.java.exception.ClientException.INVALID_ALG_PARAMETER;
-import static com.microsoft.identity.common.java.exception.ClientException.INVALID_BLOCK_SIZE;
-import static com.microsoft.identity.common.java.exception.ClientException.INVALID_KEY;
-import static com.microsoft.identity.common.java.exception.ClientException.INVALID_KEY_MISSING;
-import static com.microsoft.identity.common.java.exception.ClientException.INVALID_PROTECTION_PARAMS;
-import static com.microsoft.identity.common.java.exception.ClientException.JSON_CONSTRUCTION_FAILED;
-import static com.microsoft.identity.common.java.exception.ClientException.JWT_SIGNING_FAILURE;
-import static com.microsoft.identity.common.java.exception.ClientException.KEYSTORE_NOT_INITIALIZED;
-import static com.microsoft.identity.common.java.exception.ClientException.NO_SUCH_ALGORITHM;
-import static com.microsoft.identity.common.java.exception.ClientException.NO_SUCH_PADDING;
-import static com.microsoft.identity.common.java.exception.ClientException.SIGNING_FAILURE;
-import static com.microsoft.identity.common.java.exception.ClientException.THUMBPRINT_COMPUTATION_FAILURE;
-import static com.microsoft.identity.common.java.exception.ClientException.UNKNOWN_EXPORT_FORMAT;
-
-/**
- * Concrete class providing convenience functions around AndroidKeystore to support PoP.
- */
-@SuppressWarnings("PMD.AvoidDuplicateLiterals")
-public class DevicePopManager implements IDevicePopManager {
-
-    private static final String TAG = DevicePopManager.class.getSimpleName();
+public abstract class AbstractDevicePopManager implements IDevicePopManager {
+    private static final String TAG = AbstractDevicePopManager.class.getSimpleName();
 
     private static final Charset UTF8 = Charset.forName("UTF-8");
 
     /**
      * The PoP alias in the designated KeyStore -- default val used by non-OneAuth Android platform.
      */
-    private static final String DEFAULT_KEYSTORE_ENTRY_ALIAS = "microsoft-device-pop";
+    public static final String DEFAULT_KEYSTORE_ENTRY_ALIAS = "microsoft-device-pop";
```

Comment:
nit: protected?

Reply:
Have to keep `public` because I use it in the ProxyPopManager [here](https://github.com/AzureAD/ad-accounts-for-android/blob/1c9692907f4abc8d560ff6268fd0651d0ae06f28/LinuxBroker/src/main/com/microsoft/identity/broker/pop/ProxyDevicePopManager.java#L25) and that class is not a child of AbstractDevicePopManager.

The other option is to declare the constant separately in each file.